### PR TITLE
remove LB circular in OpenStack

### LINKED ISSUE #9620: circular dependency when creating OpenStack cluster with loadbalancer
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

master

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

**3. What cloud provider are you using?**

OpenStack

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

```
~/go/bin/kops create cluster \
  --cloud openstack \
  --name calc.k8s.local \
  --state ${KOPS_STATE_STORE} \
  --zones zone-1,zone-2,zone-3 \
  --network-cidr 10.1.0.0/16 \
  --image debian-10-260520-devops \
  --master-count=3 \
  --node-count=2 \
  --node-size m1.medium \
  --master-size m1.medium \
  --etcd-storage-type solidfire \
  --topology private \
  --bastion \
  --networking calico \
  --api-loadbalancer-type public \
  --os-kubelet-ignore-az=true \
  --os-ext-net int-net
```

**5. What happened after the commands executed?**

```
% ~/go/bin/kops create cluster \
  --cloud openstack \
  --name calc.k8s.local \
  --state ${KOPS_STATE_STORE} \
  --zones zone-1,zone-2,zone-3 \
  --network-cidr 10.1.0.0/16 \
  --image debian-10-260520-devops \
  --master-count=3 \
  --node-count=2 \
  --node-size m1.medium \
  --master-size m1.medium \
  --etcd-storage-type solidfire \
  --topology private \
  --bastion \
  --networking calico \
  --api-loadbalancer-type public \
  --os-kubelet-ignore-az=true \
  --os-ext-net int-net
I0723 22:50:03.690384   75337 featureflag.go:156] FeatureFlag "EnableExternalCloudController"=true
I0723 22:50:05.742648   75337 subnets.go:184] Assigned CIDR 10.1.32.0/19 to subnet zone-1
I0723 22:50:05.742724   75337 subnets.go:184] Assigned CIDR 10.1.64.0/19 to subnet zone-2
I0723 22:50:05.742790   75337 subnets.go:184] Assigned CIDR 10.1.96.0/19 to subnet zone-3
I0723 22:50:05.742845   75337 subnets.go:198] Assigned CIDR 10.1.0.0/22 to subnet utility-zone-1
I0723 22:50:05.742911   75337 subnets.go:198] Assigned CIDR 10.1.4.0/22 to subnet utility-zone-2
I0723 22:50:05.742966   75337 subnets.go:198] Assigned CIDR 10.1.8.0/22 to subnet utility-zone-3
I0723 22:50:08.507526   75337 create_cluster.go:693] Using SSH public key: /Users/xx/.ssh/id_rsa.pub
Previewing changes that will be made:

I0723 22:50:10.391004   75337 apply_cluster.go:423] Gossip DNS: skipping DNS validation
I0723 22:50:11.773314   75337 executor.go:103] Tasks: 0 done / 131 total; 53 can run
W0723 22:50:11.870574   75337 vfs_castore.go:604] CA private key was not found
I0723 22:50:14.414588   75337 executor.go:103] Tasks: 53 done / 131 total; 48 can run
I0723 22:50:15.866852   75337 executor.go:103] Tasks: 101 done / 131 total; 12 can run
I0723 22:50:17.304464   75337 executor.go:103] Tasks: 113 done / 131 total; 0 can run

error running tasks: Unable to execute tasks (circular dependency): PoolAssociation/calc.k8s.local-master-zone-2, FloatingIP/fip-api.calc.k8s.local, LB/api.calc.k8s.local, Instance/master-zone-1-1-calc-k8s-local, Instance/nodes-2-calc-k8s-local, Instance/bastions-1-calc-k8s-local, FloatingIP/fip-bastions-1-calc-k8s-local, LBListener/api.calc.k8s.local, Instance/master-zone-2-1-calc-k8s-local, BootstrapScript/master-zone-1, Instance/master-zone-3-1-calc-k8s-local, BootstrapScript/master-zone-3, BootstrapScript/master-zone-2, PoolAssociation/calc.k8s.local-master-zone-3, BootstrapScript/nodes, PoolAssociation/calc.k8s.local-master-zone-1, Instance/nodes-1-calc-k8s-local, LBPool/api.calc.k8s.local-https
```

/kind bug

### PR DESCRIPTION
**Pull request by zetaab**
fixes #9620

### CODE REVIEW

File: upup/pkg/fi/cloudup/openstacktasks/lb.go [hunk 1]
```diff
@@ -93,12 +93,6 @@ func (e *LB) GetDependencies(tasks map[string]fi.Task) []fi.Task {
 		if _, ok := task.(*Subnet); ok {
 			deps = append(deps, task)
 		}
-		if _, ok := task.(*ServerGroup); ok {
-			deps = append(deps, task)
-		}
-		if _, ok := task.(*Instance); ok {
```
Comment (zetaab) [inline upup/pkg/fi/cloudup/openstacktasks/lb.go:99]: ServerGroup & Instance are not used in LB task. So I do not see why these are dependencies

**Comment by johngmyers** [comment]
@zetaab You now need to pass the `--admin` flag to `kops update cluster --yes` in order for it to write a kubecfg.

**Comment by zetaab** [comment]
@johngmyers I did have similar named cluster already in kubecfg, so if I read code correctly its thinking that it do exist already. I try again with different name that does not exist in kubecfg. Hopefully that works

**Comment by zetaab** [comment]
nah it does not help, but `--admin` do work. Is there some release note about this?

File: upup/pkg/fi/cloudup/openstacktasks/lb.go (hunk 1 above)
Comment (johngmyers) [reply upup/pkg/fi/cloudup/openstacktasks/lb.go:99]: I agree. This alone should break the circular dependency.

**Comment by johngmyers** [comment]
It's the first thing mentioned under "Significant changes"

**Pull request merged**